django/contrib/admin/widgets.py

```diff
@@ -380,18 +380,18 @@ class AutocompleteMixin:
     Renders the necessary data attributes for select2 and adds the static form
     media.
     """
-    url_name = '%s:%s_%s_autocomplete'
+    url_name = '%s:autocomplete'
 
-    def __init__(self, rel, admin_site, attrs=None, choices=(), using=None):
-        self.rel = rel
+    def __init__(self, field, admin_site, attrs=None, choices=(), using=None):
```

Comment:
Hi @codingjoe. `self.choices` should be a` ModelChoiceIterator` instance, so `self.field` can be derived from `self.choices.field`. With this in mind, is the new `field` init argument really necessary?

Reply:
Not sure who marked this as resolved, because it isn't.

Reply:
AdminSelect used to take the related model. It now takes the field.